Files/ViewModels/ItemViewModel.cs

```diff
@@ -800,38 +865,71 @@ public async void RapidAddItemsToCollectionAsync(string path, string previousDir
                 semaphoreCTS = new CancellationTokenSource();
 
                 IsLoadingItems = true;
-                IsFolderEmptyTextDisplayed = false;
                 filesAndFolders.Clear();
+                await ApplyFilesAndFoldersChangesAsync();
                 Stopwatch stopwatch = new Stopwatch();
                 stopwatch.Start();
 
                 AssociatedInstance.NavigationToolbar.CanGoBack = AssociatedInstance.ContentFrame.CanGoBack;
                 AssociatedInstance.NavigationToolbar.CanGoForward = AssociatedInstance.ContentFrame.CanGoForward;
 
-                var cacheEntry = await fileListCache.ReadFileListFromCache(path);
-                if (cacheEntry != null)
+                (bool LoadedFromCache, ListedItem CurrentFolder) cacheResult = (false, null);
+
+                if (useCache)
                 {
-                    CurrentFolder = cacheEntry.CurrentFolder;
-                    var orderedList = OrderFiles2(cacheEntry.FileList);
-                    OrderFiles(orderedList);
-                    Debug.WriteLine($"Loading of items from cache in {WorkingDirectory} completed in {stopwatch.ElapsedMilliseconds} milliseconds.\n");
-                    IsLoadingIndicatorActive = false;
+                    cacheResult = await Task.Run(async () =>
```

Comment:
So if I understand correctly - we are waiting for results to load from cache and only after that we're loading fresh data from file system? I think that's totally fine for small folders, but on large ones that might make everything longer. I wonder if maybe cache should be for small folders only? Or maybe only first 300 items should be cached, even if folder is bigger? @gave92 @yaichenbaum what do you think about that?

Or maybe there's a way to load cache and filesystem in parallel? Although this might be really hard to synchronize. 
I think we should define equality check for 'ListedItem' so that we can compare items from filesystem and cache effectively (right now comparison is done by reference, which will be different when loading from sqlite)

Reply:
In my experiment, loading from the persistent cache for large folders is slow, and the user experience is lower than loading them from filesystem directly ("loading many entries at once but takes several seconds" vs "gradually loading them from filesystem but some entries can be immediately listed").
However, the memory cache does speed up a lot in both small and large folder cases, so I committed f76c98c which loads persistent cache into memory cache if memory cache is absent.

Reply:
Ok, so that leaves only one issue - the first time load of a big folders from sqlite might be slower then gradually loading them from filesystem. Hence my proposal for limiting cache only for smaller folders or to cache only first 32 files for initial user interaction and loading the rest from filesystem. I believe that's a decision for maintainers

Reply:
> Ok, so that leaves only one issue - the first time load of a big folders from sqlite might be slower then gradually loading them from filesystem. Hence my proposal for limiting cache only for smaller folders or to cache only first 32 files for initial user interaction and loading the rest from filesystem. I believe that's a decision for maintainers

I like the idea of loading only the first batch from the cache on the first interaction.

Reply:
So we can just cache first 32 items (or less) and ignore the rest. The gradual loading is fast enough that user won't be able to "go through" cached items while new ones will appear from filesystem. @hez2010 Can you add that in this PR?

Reply:
Done in 06b47f4.

Reply:
Awesome!